api/pkg/apis/v1alpha1/utils/apiclient.go

```diff
@@ -0,0 +1,614 @@
+		QueueJob(ctx context.Context, id string, namespace string, isDelete bool, isTarget bool) error
+		QueueDeploymentJob(ctx context.Context, namespace string, isDelete bool, deployment model.DeploymentSpec) error
+	}
+
+	ApiClient interface {
+		SummaryGetter
+		Dispatcher
+		GetInstancesForAllNamespaces(ctx context.Context) ([]model.InstanceState, error)
+		GetInstances(ctx context.Context, namespace string) ([]model.InstanceState, error)
+		GetInstance(ctx context.Context, instance string, namespace string) (model.InstanceState, error)
+		CreateInstance(ctx context.Context, instance string, payload []byte, namespace string) error
+		DeleteInstance(ctx context.Context, instance string, namespace string) error
+		DeleteTarget(ctx context.Context, target string, namespace string) error
+		GetSolutions(ctx context.Context, namespace string) ([]model.SolutionState, error)
+		GetSolution(ctx context.Context, solution string, namespace string) (model.SolutionState, error)
+		CreateSolution(ctx context.Context, solution string, payload []byte, namespace string) error
+		DeleteSolution(ctx context.Context, solution string, namespace string) error
+		GetTargetsForAllNamespaces(ctx context.Context) ([]model.TargetState, error)
+		GetTarget(ctx context.Context, target string, namespace string) (model.TargetState, error)
+		GetTargets(ctx context.Context, namespace string) ([]model.TargetState, error)
+		CreateTarget(ctx context.Context, target string, payload []byte, namespace string) error
+		Reconcile(ctx context.Context, deployment model.DeploymentSpec, isDelete bool, namespace string) (model.SummarySpec, error)
+	}
+)
+
+func noTokenProvider(ctx context.Context, baseUrl string, client *http.Client) (string, error) {
+	return "", nil
+}
+
+func WithUserPassword(ctx context.Context, user string, password string) ApiClientOption {
+	return func(a *apiClient) {
+		a.tokenProvider = func(ctx context.Context, baseUrl string, _ *http.Client) (string, error) {
+			request := authRequest{Username: user, Password: password}
+			requestData, _ := json.Marshal(request)
+			ret, err := a.callRestAPI(ctx, "users/auth", "POST", requestData, "")
+			if err != nil {
+				return "", err
+			}
+
+			var response authResponse
+			err = json.Unmarshal(ret, &response)
+			if err != nil {
+				return "", err
+			}
+
+			return response.AccessToken, nil
+		}
+	}
+}
+
+func WithServiceAccountToken() ApiClientOption {
+	return func(a *apiClient) {
+		a.tokenProvider = func(ctx context.Context, _ string, _ *http.Client) (string, error) {
+			path := os.Getenv(constants.SATokenPathName)
+			if path == "" {
+				path = constants.SATokenPath
+			}
+			token, err := os.ReadFile(path)
+			if err != nil {
+				return "", v1alpha2.NewCOAError(nil, "Token creation error: unable to read from volume.", v1alpha2.InternalError)
+			}
+			return string(token), nil
+		}
+	}
+}
+
+func WithCertAuth(caCertPath string) ApiClientOption {
+	return func(a *apiClient) {
+		a.caCertPath = caCertPath
```

Comment:
Auth by cert is not supported yet right? Looks like only token based auth is supported in this file.

Reply:
No, auth by cert is not supported yet. This is just exposing a api client options.